src/librustc_trans/cabi_x86_64.rs

```diff
@@ -11,388 +11,260 @@
 // The classification code for the x86_64 ABI is taken from the clay language
 // https://github.com/jckarter/clay/blob/master/compiler/src/externals.cpp
 
-#![allow(non_upper_case_globals)]
-use self::RegClass::*;
-
-use llvm::{Integer, Pointer, Float, Double};
-use llvm::{Struct, Array, Vector};
-use abi::{self, ArgType, ArgAttribute, FnType};
+use abi::{ArgType, ArgAttribute, FnType, LayoutExt, Reg, RegKind};
 use context::CrateContext;
-use type_::Type;
 
-#[derive(Clone, Copy, PartialEq)]
-enum RegClass {
-    NoClass,
+use rustc::ty::layout::{self, Layout, TyLayout, Size};
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+enum Class {
+    None,
     Int,
-    SSEFs,
-    SSEFv,
-    SSEDs,
-    SSEDv,
-    SSEInt(/* bitwidth */ u64),
+    LoneF32,
+    Sse,
```

Comment:
There's a more elegant way to handle this (last `Sse` with only 4 bytes of data left, only way to get `LoneF32` anyway, unless you have weird padding). I'll try to implement it.
Also, clang does somewhat complex type recovery - I don't think we care, as this is for FFI.